mirascope/core/anthropic/stream.py

```diff
@@ -34,9 +40,67 @@ def cost(self) -> float | None:
     def _construct_message_param(
         self, tool_calls: list[ToolUseBlock] | None = None, content: str | None = None
     ) -> MessageParam:
-        """Returns the tool message parameters for tool call results."""
+        """Constructs the message parameter for the assistant."""
+        message_param_tool_calls = []
+        if tool_calls:
+            message_param_tool_calls: list[ToolUseBlockParam] = [
+                tool_call.model_dump()
+                for tool_call in tool_calls  # type: ignore
+            ]
         return {
             "role": "assistant",
-            "content": ([TextBlock(text=content, type="text")] if content else [])
-            + (tool_calls or []),
+            "content": ([TextBlockParam(text=content, type="text")] if content else [])
+            + message_param_tool_calls,
         }
+
+    def construct_call_response(self) -> AnthropicCallResponse:
```

Comment:
missing docstring for this function. Make sure in the description to make it clear that this is a constructed response and might not be 1:1 equivalent with what the response would be if you made the same call without streaming

Reply:
done